src/bool.rs

```diff
@@ -120,12 +157,25 @@ pub fn guarded_transmute_bool_vec_permissive(bytes: Vec<u8>) -> Result<Vec<bool>
 /// # run().unwrap()
 /// ```
 #[cfg(feature = "std")]
-pub fn guarded_transmute_bool_vec_pedantic(bytes: Vec<u8>) -> Result<Vec<bool>, Error> {
+pub fn safe_transmute_bool_vec_pedantic(bytes: Vec<u8>) -> Result<Vec<bool>, Error> {
     check_bool(&bytes)?;
-    unsafe { guarded_transmute_vec_pedantic(bytes) }
+    // alignment guarantees are ensured, and all values have been checked,
+    // so the conversion is safe.
+    unsafe { guarded_transmute_vec::<_, PedanticGuard>(bytes) }
 }
 
+/// Transform a byte vector into a vector of bool.
+///
+/// The vector's allocated byte buffer will be reused when possible, and
+/// should not have extraneous data.
+#[cfg(feature = "std")]
+#[deprecated(since = "0.11.0", note = "use `safe_transmute_bool_vec_pedantic` instead")]
+pub fn guarded_transmute_bool_vec_pedantic(bytes: Vec<u8>) -> Result<Vec<bool>, Error> {
+    safe_transmute_bool_vec_pedantic(bytes)
+}
 
+/// Helper function for raising an error if any of the bytes does not make a
+/// valid `bool`.
 fn check_bool(bytes: &[u8]) -> Result<(), Error> {
```

Comment:
Just making sure: this isn't supposed to be `pub` despite the doccomment, right?

Reply:
It is indeed private, but I felt like documenting it nonetheless.

Reply:
👌